src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.CoreCLR.cs

```diff
@@ -168,44 +169,51 @@ public override int GetHashCode()
 
         public override bool Equals(object? obj)
         {
-            if (!IsGenericMethod)
-                return obj == (object)this;
+            if (IsGenericMethod)
```

Comment:
There is nothing changed within IsGenericMethod true case, just reverted the conditional